[Core][flaky-test] linux://python/ray/tests:test_scheduling_debug_mode is flaky

### What happened + What you expected to happen

test_locality_aware_leasing_borrowed_objects failed with following errors.

```


=================================== FAILURES ===================================
--
  | _________________ test_locality_aware_leasing_borrowed_objects _________________
  |  
  | ray_start_cluster = <ray.cluster_utils.Cluster object at 0x7ff17c027780>
  |  
  | def test_locality_aware_leasing_borrowed_objects(ray_start_cluster):
  | # This test ensures that a task will run where its task dependencies are
  | # located, even when those objects are borrowed.
  | cluster = ray_start_cluster
  |  
  | # Disable worker caching so worker leases are not reused, and disable
  | # inlining of return objects so return objects are always put into Plasma.
  | cluster.add_node(
  | num_cpus=1,
  | resources={"pin_head": 1},
  | _system_config={
  | "worker_lease_timeout_milliseconds": 0,
  | "max_direct_call_object_size": 0,
  | },
  | )
  | # Use a custom resource for pinning tasks to a node.
  | worker_node = cluster.add_node(num_cpus=1, resources={"pin_worker": 1})
  | ray.init(address=cluster.address)
  |  
  | @ray.remote
  | def f():
  | return ray.worker.global_worker.node.unique_id
  |  
  | @ray.remote
  | def g(x):
  | return ray.get(h.remote(x[0]))
  |  
  | @ray.remote
  | def h(x):
  | return ray.worker.global_worker.node.unique_id
  |  
  | # f will run on worker, f_obj will be pinned on worker.
  | f_obj = f.options(resources={"pin_worker": 1}).remote()
  | # g will run on head, f_obj will be borrowed by head, and we confirm that
  | # h(f_obj) is scheduled onto worker, the node that has f_obj.
  | >       assert (
  | ray.get(g.options(resources={"pin_head": 1}).remote([f_obj]))
  | == worker_node.unique_id
  | )
  | E       AssertionError: assert '172.18.0.3:/.../plasma_store' == '172.18.0.3:/...lasma_store.1'
  | E         - 172.18.0.3:/tmp/ray/session_2022-04-17_04-15-32_713148_13164/sockets/plasma_store.1
  | E         ?                                                                                  --
  | E         + 172.18.0.3:/tmp/ray/session_2022-04-17_04-15-32_713148_13164/sockets/plasma_store
  |  
  | /ray/python/ray/tests/test_scheduling.py:418: AssertionError
  | ----------------------------- Captured stderr call -----------------------------
  | 2022-04-17 04:15:34,961	INFO services.py:1466 -- View the Ray dashboard at http://127.0.0.1:8265
  | 2022-04-17 04:15:35,948	INFO worker.py:948 -- Connecting to existing Ray cluster at address: 172.18.0.3:63248
  | =============================== warnings summary ===============================
  | /opt/miniconda/lib/python3.6/site-packages/_pytest/config/__init__.py:935
  | /opt/miniconda/lib/python3.6/site-packages/_pytest/config/__init__.py:935: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: anyio
  | self._mark_plugins_for_rewrite(hook)
  |  


```

### Versions / Dependencies

latest ray

### Reproduction script

buildkite

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

--- a/python/ray/tests/test_scheduling.py
+++ b/python/ray/tests/test_scheduling.py
@@ -413,6 +413,10 @@ def test_locality_aware_leasing_borrowed_objects(ray_start_cluster):
 
     # f will run on worker, f_obj will be pinned on worker.
     f_obj = f.options(resources={"pin_worker": 1}).remote()
+    # Make sure owner has the location information for f_obj,
+    # before we launch g so g worker can get the locality information
+    # from the owner.
+    ray.wait([f_obj], fetch_local=False)
     # g will run on head, f_obj will be borrowed by head, and we confirm that
     # h(f_obj) is scheduled onto worker, the node that has f_obj.
     assert (

